core/consensus/babe/src/lib.rs

```diff
@@ -640,6 +646,33 @@ impl<B: Block, C> Verifier<B> for BabeVerifier<C> where
 					fork_choice: ForkChoiceStrategy::LongestChain,
 				};
 
+				// We do not produce an `Instant` synthetically, so this cannot panic.
+				let slot_num: u32 = u32::try_from(slot_num)
+					.expect("FIXME: We cannot handle slot numbers above u32::MAX yet!");
+				let mut timestamps = self.timestamps.lock();
+				// Remainder of this block is a critical section.
+				let num_timestamps = timestamps.1.len();
+				if num_timestamps as u64 >= self.config.0.median_required_blocks {
+					let mut new_list: Vec<_> = timestamps.1.iter().enumerate().map(|(_counter, &(t, sl)): (usize, &(Instant, u64))| {
+						t + Duration::from_secs(self.config.get())
```

Comment:
`SlotDuration::slot_duration` returns an `u64` as millis, is it really correct with to construct with `Duration::from_secs`?!